operator/v1/types_ingress.go

```diff
@@ -282,6 +287,230 @@ var (
+	//   ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:
+	//   DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:
+	//   ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:
+	//   AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS
+	// securityProtocol:
+	//   minimumVersion: TLSv1.0
+	//   maximumVersion: TLSv1.2
+	// dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Intermediate *IntermediateSecurityProfile `json:"intermediate,omitempty"`
+	// modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	// and looks like this:
+	//
+	// cipers: >-
+	//   ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:
+	//   ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:
+	//   ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256
+	// securityProtocol:
+	//   minimumVersion: TLSv1.2
+	//   maximumVersion: TLSv1.2
+	// dhParamSize: 2048
+	//
+	// +optional
+	// +nullable
+	Modern *ModernSecurityProfile `json:"modern,omitempty"`
+	// custom is a user-defined security profile. An example custom profile
+	// looks like this:
+	//
+	// cipers: >-
+	//   ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:
+	//   ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256
+	// securityProtocol:
+	//   minimumVersion: TLSv1.1
+	//   maximumVersion: TLSv1.2
+	// dhParamSize: 1024
+	//
+	// +optional
+	// +nullable
+	Custom *CustomSecurityProfile `json:"custom,omitempty"`
+}
+
+type OldSecurityProfile struct{}
+type IntermediateSecurityProfile struct{}
+type ModernSecurityProfile struct{}
+
+type CustomSecurityProfile struct {
+	// securityProfile defines the schema for a custom security profile.
+	//
+	// Any unspecified fields will default to the "Intermediate" compatibility configuration
+	// as defined by:
+	//
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	SecurityProfile `json:"securityProfile"` // inline
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// Old is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// Intermediate is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// Modern is a security profile based on:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	Modern SecurityProfileType = "Modern"
+	// Custom is a security profile that allows for user-defined parameters.
+	Custom SecurityProfileType = "Custom"
```

Comment:
You have the `SecurityProfile` prefix in `SecurityProfileOldType` and `SecurityProfileIntermediateType` and omit it from `Modern` and `Custom`.  The names should be consistent.

Reply:
ack